packages/sats-wagmi/src/hooks/useBalance.tsx

```diff
@@ -29,10 +30,34 @@ const useBalance = (props: UseBalanceProps = {}) => {
       }
 
       const esploraClient = new EsploraClient(network);
+      const ordinalsClient = new OrdinalsClient(network);
 
-      const { confirmed, unconfirmed, total } = await esploraClient.getBalance(address);
+      const addressInfo = getAddressInfo(address);
 
-      return { confirmed: BigInt(confirmed), unconfirmed: BigInt(unconfirmed), total: BigInt(total) };
+      if (addressInfo.type === AddressType.p2tr) {
+        const [{ confirmed, unconfirmed, total }, inscribed, runic] = await Promise.all([
+          esploraClient.getBalance(address),
+          ordinalsClient.getOutputsFromAddress(address, 'inscribed'),
+          ordinalsClient.getOutputsFromAddress(address, 'runic')
+        ]);
+
+        const inscribedOutputsTotal = inscribed.reduce((acc, output) => acc + output.value, 0);
+        const runicOutputsTotal = runic.reduce((acc, output) => acc + output.value, 0);
+
+        return {
+          confirmed: BigInt(confirmed - inscribedOutputsTotal - runicOutputsTotal),
+          unconfirmed: BigInt(unconfirmed),
+          total: BigInt(total - inscribedOutputsTotal - runicOutputsTotal)
```

Comment:
if `ord` is indexing more slowly than `esplora` then this can be incorrect. A safer way would be to use an approach that corresponds more closely to the feeestimation/signing logic:

- use the esplora `/address/:address/utxo` endpoint to get all utxos
- use the ord `getOutputsFromAddress(address, 'cardinal')` function and filter out any utxos found above that are not returned by ord
- calculate the confirmed, unconfirmed and total balances based on the utxos (the item returned by esplora contains a `confirmed` field that can be used for this)

I think we still don't know if ord returns unconfirmed utxos. If it doesn't, then `unconfirmed` will be made 0 with my approach, which is not ideal, but at least it's consistent with the signing logic

Reply:
From ord codebase changelog
```- Don't consider unconfirmed UTXOs as spent ([#3255](https://github.com/ordinals/ord/pull/3255) by [arik-so](https://github.com/arik-so))```

rpc call - https://developer.bitcoin.org/reference/rpc/gettxout.html
https://github.com/ordinals/ord/blob/e2b0732cb78b44d9c4edc92f82237ab1affdff04/src/subcommand/server.rs#L748

what I got from quickly looking at that code is that ord treats utxo in mempoll as not spent (reflected in the title of that change)